src/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -446,8 +441,9 @@ public virtual void Emit(OpCode opcode, short arg)
             // Puts opcode onto the stream of instructions followed by arg
             EnsureCapacity(5);
             InternalEmit(opcode);
-            m_ILStream[m_length++] = (byte)arg;
-            m_ILStream[m_length++] = (byte)(arg >> 8);
+            m_ILStream[m_length]     = (byte)arg;
```

Comment:
Why not use `WriteInt16LittleEndian` here?

Reply:
Benchmarked with Span and BinaryPrimitives, but this was faster, as the overhead of creating a span it lasger than just doing the aridhmetic operations.

Reply:
Did some confirmatory benchmarks, but same thing, still slight performance regressions.